compiler/rustc_const_eval/src/transform/validate.rs

```diff
@@ -557,7 +557,14 @@ impl<'a, 'tcx> Visitor<'tcx> for TypeChecker<'a, 'tcx> {
             }
             Rvalue::Cast(kind, operand, target_type) => {
                 match kind {
-                    CastKind::Misc => {
+                    CastKind::DynStar => {
+                        // FIXME(dyn-star): make sure nothing needs to be done here.
+                    }
+                    // Nothing to check here
+                    CastKind::PointerFromExposedAddress
+                    | CastKind::PointerExposeAddress
+                    | CastKind::Pointer(_) => {}
+                    _ => {
```

Comment:
We should check that the types are as expected here (leaving out `CastKind::Pointer(_)` might be fine). This will probably need some deduplication with the typeck code

Reply:
I am not sure what to check 😅 , with new function `mir_cast_kind` we already report a bug with any unexpected cast, and since we built `Rvalue`s with that new function is a check really needed?

Reply:
Yes, because MIR validation doesn't just run after MIR is built, but also after transformations that modify it

Reply:
I think I modified those too, but it's better to be safe than sorry will add the checks in the next PR.